src/identity/interaction/InteractionHandler.ts

```diff
@@ -1,43 +1,37 @@
 import type { KoaContextWithOIDC } from 'oidc-provider';
 import type { Operation } from '../../http/Operation';
+import type { Representation } from '../../http/representation/Representation';
 import { APPLICATION_JSON } from '../../util/ContentTypes';
 import { NotImplementedHttpError } from '../../util/errors/NotImplementedHttpError';
 import { AsyncHandler } from '../../util/handlers/AsyncHandler';
 
 // OIDC library does not directly export the Interaction type
 export type Interaction = NonNullable<KoaContextWithOIDC['oidc']['entities']['Interaction']>;
 
+export interface InteractionOperation extends Operation {
+  method: 'GET' | 'POST';
+}
+
 export interface InteractionHandlerInput {
   /**
-   * The operation to execute
+   * The operation to execute. Only GET and POST operations are supported.
    */
-  operation: Operation;
+  operation: InteractionOperation;
   /**
    * Will be defined if the OIDC library expects us to resolve an interaction it can't handle itself,
    * such as logging a user in.
    */
   oidcInteraction?: Interaction;
 }
 
-export type InteractionHandlerResult = InteractionResponseResult | InteractionErrorResult;
-
-export interface InteractionResponseResult<T = NodeJS.Dict<any>> {
-  type: 'response';
-  details?: T;
-}
-
-export interface InteractionErrorResult {
-  type: 'error';
-  error: Error;
-}
-
 /**
  * Handler used for IDP interactions.
  * Only supports JSON data.
  */
-export abstract class InteractionHandler extends AsyncHandler<InteractionHandlerInput, InteractionHandlerResult> {
+export abstract class InteractionHandler extends AsyncHandler<InteractionHandlerInput, Representation> {
   public async canHandle({ operation }: InteractionHandlerInput): Promise<void> {
```

Comment:
Should indeed probably check for the correct method here.

Reply:
I'm actually going to change this handler and the `IdentityProviderHttpHandler` so they accept all methods, and only the `BaseInteractionHandler` is going to support GET and POST. No need to prevent other options in the future.